### ISSUE TITLE
[Android][Bug] Restore MapView when location permission is not granted anymore

### ISSUE THREAD

**Issue by mayardb** [mapbox/mapbox-gl-native#3462]
With Marshmallow, user can deny location permission.
Every works fine, with the sample excepting the restoration

### How to:
- Accept the location permission
- Enable the gps
- Pause the application and refuse the location permission
- Resume the application

### Observed:
The application crashes when restoring the MapView

#### Expected:
The application does not crash and the locationFAB is disabled.

#### StackTrace : 
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.mapbox.mapboxsdk.testapp/com.mapbox.mapboxsdk.testapp.MainActivity}: java.lang.SecurityException: "passive" location provider requires ACCESS_FINE_LOCATION permission.
                                                                                at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
                                                                                at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
                                                                                at android.app.ActivityThread.-wrap11(ActivityThread.java)
                                                                                at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
                                                                                at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                at android.os.Looper.loop(Looper.java:148)
                                                                                at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                                at java.lang.reflect.Method.invoke(Native Method)
                                                                                at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                                at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                                             Caused by: java.lang.SecurityException: "passive" location provider requires ACCESS_FINE_LOCATION permission.
                                                                                at android.os.Parcel.readException(Parcel.java:1599)
                                                                                at android.os.Parcel.readException(Parcel.java:1552)
                                                                                at android.location.ILocationManager$Stub$Proxy.getLastLocation(ILocationManager.java:717)
                                                                                at android.location.LocationManager.getLastKnownLocation(LocationManager.java:1200)
                                                                                at com.mapzen.android.lost.internal.FusionEngine.getLastLocation(FusionEngine.java:50)
                                                                                at com.mapzen.android.lost.internal.FusedLocationProviderApiImpl.getLastLocation(FusedLocationProviderApiImpl.java:32)
                                                                                at com.mapbox.mapboxsdk.location.LocationServices.toggleGPS(LocationServices.java:66)
                                                                                at com.mapbox.mapboxsdk.views.UserLocationView.toggleGps(UserLocationView.java:397)
                                                                                at com.mapbox.mapboxsdk.views.UserLocationView.setEnabled(UserLocationView.java:267)
                                                                                at com.mapbox.mapboxsdk.views.MapView.setMyLocationEnabled(MapView.java:3875)
                                                                                at com.mapbox.mapboxsdk.views.MapView.onCreate(MapView.java:922)
                                                                                at com.mapbox.mapboxsdk.testapp.MainActivity.onCreate(MainActivity.java:117)
                                                                                at android.app.Activity.performCreate(Activity.java:6237)
                                                                                at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
                                                                                at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369)
                                                                                at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476) 
                                                                                at android.app.ActivityThread.-wrap11(ActivityThread.java) 
                                                                                at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344) 
                                                                                at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                                at android.os.Looper.loop(Looper.java:148) 
                                                                                at android.app.ActivityThread.main(ActivityThread.java:5417) 
                                                                                at java.lang.reflect.Method.invoke(Native Method) 
                                                                                at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
                                                                                at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)

**Comment by tobrun** [comment]
@mayardb thank you for reaching out, I integrated support annotations yesterday to enforce the checks for Marshmallow permissions, today I made it a bit more robust by catching SecurityExceptions and ignoring them https://github.com/mapbox/mapbox-gl-native/issues/3435. 

If the user now revokes the permission and returns to the app it will not show the user location and not crash anymore. In the end it will be the responsibility of the end developer to handle permissions checks and not the library itself. (we will not handle results coming from `onRequestPermissionsResult`).
